filebeat/module/kibana/audit/ingest/pipeline-json.yml

```diff
@@ -16,15 +16,15 @@ processors:
 - set:
     if: ctx.kibana._audit_temp.event.category != null
     field: event.category
-    value: "{{kibana._audit_temp.event.category}}"
+    value: "{{kibana._audit_temp.event.category.0}}"
 - set:
     if: ctx.kibana._audit_temp.event.outcome != null
     field: event.outcome
     value: "{{kibana._audit_temp.event.outcome}}"
 - set:
     if: ctx.kibana._audit_temp.event.type != null
     field: event.type
-    value: "{{kibana._audit_temp.event.type}}"
```

Comment:
In order to preserve backwards compatibility, we'd probably need to first check if the field is an array and only if it is, pluck out the first value, otherwise use it as is. 

However, I'm wondering what our strategy is. This ingest pipeline only handles audit logs but with your change in https://github.com/elastic/kibana/pull/96350 we would probably want to use it for all Kibana logs. 

For audit logs specifically, we'd want to ingest ECS category and type fields as arrays moving forwards but can't change this straight away since it could break existing filters. What's our migration strategy for these and do we know if all apps in Kibana handle arrays correctly (Logs app as you pointed out currently does not).

Reply:
> In order to preserve backwards compatibility, we'd probably need to first check if the field is an array and only if it is, pluck out the first value, otherwise use it as is.

Good point, will update.

> This ingest pipeline only handles audit logs but with your change in elastic/kibana#96350 we would probably want to use it for all Kibana logs.

I only changed the format of the audit logs in that PR though (TS didn't complain about any other usages of `category` and `type` outside of audit logs), which is why I only updated the audit ingest pipeline here. 

> For audit logs specifically, we'd want to ingest ECS category and type fields as arrays moving forwards but can't change this straight away since it could break existing filters. What's our migration strategy for these and do we know if all apps in Kibana handle arrays correctly (Logs app as you pointed out currently does not).

I opened https://github.com/elastic/beats/issues/24136 awhile ago for tracking the work that needs to happen to prepare the module for breaking changes in 8.0. I will take a look at how other apps are handling arrays though, and open issues for those teams accordingly.